re.go

```diff
@@ -29,6 +29,41 @@ import (
 	"strconv"
 )
 
+// Range is a special type designed to be passed via pointer to Scan. When
+// passed as the third parameter to Scan, Scan will record position information
+// for the match.  This can be used in a loop to continue scanning from the end
+// of previous first match.
+type Range struct {
+	Start int
+	End   int
+}
+
+// NotFoundError is the error returned by re.Scan when
+// the regexp didn't match the string.
+type NotFoundError struct {
+	re    *regexp.Regexp
+	input []byte
+}
+
+func (e NotFoundError) Error() string {
+	return fmt.Sprintf(`re.Scan: could not find "%s" in "%s"`,
+		e.re, e.input)
+}
+
+// ParseError is the error retured by re.Scan when a submatch couldn't be
+// parsed in the format expected by the corresponding submatch output.
```

Comment:
I think we should leave ParseError out. When a user provides a parsing function they will return their own error and there will be a resulting confusion as to whether or not the error is theirs or a ParseError.

I think just adding NotFoundError is sufficient for now.

Reply:
Done.